cnd/src/config/validation.rs

```diff
@@ -0,0 +1,50 @@
+use crate::swap_protocols::ledger::ethereum::ChainId;
+use bitcoin::Network;
+use thiserror::Error;
+
+use crate::btsieve::{
+    bitcoin::{chain_info, BitcoindConnector},
+    ethereum::{net_version, Web3Connector},
+};
+
+#[derive(Error, Debug)]
+pub enum ConfigValidationError<T: std::fmt::Debug> {
+    #[error("Connected network does not match network specified in settings (expected {connected_network:?}, got {specified_network:?})")]
+    ConnectedNetworkDoesNotMatchSpecified {
+        connected_network: T,
+        specified_network: T,
+    }
+}
+
+#[derive(Debug)]
+pub enum NetworkValidationResult<T> {
+    Valid,
+    Invalid {
+        connected_network: T,
+        specified_network: T,
+    }
+}
+
+pub async fn validate_ethereum_chain_id(connection: &Web3Connector, specified: ChainId) -> Result<NetworkValidationResult<ChainId>, anyhow::Error> {
```

Comment:
I can replace this and the function below with a generic version which takes the function to fetch the network/chain id as a parameter